src/contractors.jl

```diff
@@ -28,139 +28,186 @@ function (contractor::Bisection)(X, tol)
 end
 
 """
-    newtonlike_contract(op, X, tol)
+    safe_isempty(X)
+
+Similar to `isempty` function for `IntervalBox`, but also works for `SVector`
+of `Interval`.
+"""
+safe_isempty(X) = isempty(IntervalBox(X))
 
-    Contraction operation for contractors using the first derivative of the
-    function. This contraction use a bisection scheme to refine the intervals
-    with `:unkown` status.
 
-    Currently `Newton` and `Krawczyk` contractors uses this.
 """
-function newtonlike_contract(op, C, X, tol)
-    # use Bisection contractor for this:
-    if !(contains_zero(C.f(X)))
-        return :empty, X
-    end
+    determine_root_status(contract, f, X)
 
-    # given that have the Jacobian, can also do mean value form
+Contraction operation for contractors using the first derivative of the
+function.
+
+Currently `Newton` and `Krawczyk` contractors uses this.
+"""
+function determine_root_status(op, f, X)
+    imX = f(X)
+
+    !(contains_zero(imX)) && return :empty, X
+
+    safe_isempty(imX) && return :empty, X  # X is fully outside of the domain of f
+
+    contracted_X = op(X)
 
-    NX = op(C.f, C.f′, X) ∩ X
+    # Only happens if X is partially out of the domain of f
+    safe_isempty(contracted_X) && return :unknown, X  # force bisection
+
+    # given that have the Jacobian, can also do mean value form
+    NX = contracted_X ∩ X
 
-    isempty(NX) && return :empty, X
     isinf(X) && return :unknown, NX  # force bisection
+    safe_isempty(NX) && return :empty, X
 
-    if NX ⪽ X  # isinterior; know there's a unique root inside
-        NX =  refine(X -> op(C.f, C.f′, X), NX, tol)
-        return :unique, NX
-    end
+    NX ⪽ X  && return :unique, NX  # isinterior; know there's a unique root inside
 
     return :unknown, NX
 end
 
 """
     Newton{F, FP} <: Contractor{F}
 
-    Contractor type for the Newton method.
+Contractor type for the Newton method.
 
-    # Fields
-        - `f::F`: function whose roots are searched
-        - `f::FP`: derivative or jacobian of `f`
+# Fields
+    - `f::F`: function whose roots are searched
+    - `f::FP`: derivative or jacobian of `f`
 """
 struct Newton{F,FP} <: Contractor{F}
     f::F
     f′::FP   # use \prime<TAB> for ′
 end
 
-function (C::Newton)(X, tol)
-    newtonlike_contract(𝒩, C, X, tol)
-end
+"""
+    (C::Newton)(X, tol, α=where_bisect)
 
+Contract an interval `X` using Newton operator and return the contracted
+interval together with its status.
 
+# Inputs
+    - `X`: Interval to contract.
+    - `tol`: Precision to which unique solutions are refined.
+    - `α`: Point of bisection of intervals.
 """
-Single-variable Newton operator
-"""
-function 𝒩(f, X::Interval{T}) where {T}
-    m = Interval(mid(X, where_bisect))
-
-    m - (f(m) / ForwardDiff.derivative(f, X))
+function (C::Newton)(X, tol, α=where_bisect)
+    op = x -> 𝒩(C.f, C.f′, x, α)
+    rt = determine_root_status(op, C.f, X)
+    return refine(op, rt, tol)
```

Comment:
The root status can always be determined, in the worst case it will be `:unkown`. The name should probably read `determine_region_status` though, as they me be no roots in `X`.